prefect_gcp/cloud_storage.py

```diff
@@ -1140,3 +1145,77 @@ async def upload_from_folder(
         if num_uploaded == 0:
             self.logger.warning(f"No files were uploaded from {from_folder}.")
         return bucket_folder
+
+    @sync_compatible
+    async def upload_from_dataframe(
+        self,
+        df: DataFrame,
+        to_path: str,
+        output_format: Literal["parquet", "csv"] = "csv",
+        compression: Optional[str] = None,
```

Comment:
What are your thoughts on combining these two keywords into enums?
 
```
class Compression(Enum):
    GZIP = ".gz"

class OutputFormat(Enum):
    CSV_UNCOMPRESSED = ("csv", None)
    CSV_GZIP = ("csv", Compression.GZIP)
     ...
```

`output_format: OutputFormat = OutputFormat.CSV`

Or is this overengineering?

Reply:
it's actually much safer than letting people play with the numerous permutations they can do between format and compression. Like, `.csv` can't do `snappy`, as `.parquet`can't do .bz2, 

And actually scales if for the other compression files that .to_csv can do, besides those other methods like `to_pickle`.